crates/bevy_ecs/src/query/state.rs

```diff
@@ -40,6 +40,27 @@ impl<Q: WorldQuery, F: WorldQuery> FromWorld for QueryState<Q, F> {
     }
 }
 
+impl<Q: WorldQuery, F: WorldQuery> QueryState<Q, F> {
+    pub(crate) fn as_immut(&self) -> &QueryState<Q::ReadOnly, F::ReadOnly> {
+        // Safety: invariant on `WorldQuery` trait upholds that `Q::ReadOnly` and `F::ReadOnly`
+        // have a subset of the access, and match the same archetypes/tables as `Q`/`F` respectively.
+        unsafe { self.as_whatever::<Q::ReadOnly, F::ReadOnly>() }
+    }
+
+    /// # Safety
+    ///
+    /// `NewQ` must have a subset of the access that `Q` does and match the same archetypes/tables
```

Comment:
```suggestion
    /// `NewQ` must have a subset of the access that `Q` does and match exactly the same archetypes/tables
```

Reply:
If you agree, change elsewhere too. I just want to be sure it's clear that you can't use a subset for the second part.